### PR TITLE
Added some info about append in QFS client reference

### PR DESCRIPTION
**Pull request by kstinsonqc**
Due to several requests, I added some into about append to the QFS Client Reference page.

It is meant as more of an overview but it would have prevented some confusion with some users.

Please feel free to amend it if needed

@mikeov @mckurt

### CODE REVIEW

File: wiki/QFS-Client-Reference.md [hunk 1]
```diff
@@ -283,3 +283,92 @@ _write-behind threshold_ by subsequent write calls or until user calls
...
+    int fd = client->Open(filename, O_CREAT | O_EXCL);
+    client->Write(fd, &data[0], 1);
+    client->Close(fd);
+    
+    fd = client->Open(filename, O_WRONLY | O_APPEND);
+    client->RecordAppend(fd, &data[1], 1);
+    client->RecordAppend(fd, &data[2], 1);
+    client->Close(fd);
+
+will result in a file with 2 chunks, each replicated 3 times, and with an size of
```
Comment (mckurt) [inline wiki/QFS-Client-Reference.md]: "with an size" --> "with a size"

File: wiki/QFS-Client-Reference.md [hunk 2]
```diff
@@ -283,3 +283,92 @@ _write-behind threshold_ by subsequent write calls or until user calls
...
+## Append Operations
+
+### RecordAppend
+
+RecordAppend can be used by a single writer to append to a replicated file. There is
+limited support for append to RS files.
+
+The file should be opened using the mode O_APPEND or O_WRONLY | O_APPEND.
+When RecordAppend is used to write a record, the entire record is guaranteed to be
+written to the same chunk. If the record will not fit in the current chunk, a new chunk
```
Comment (mckurt) [inline wiki/QFS-Client-Reference.md]: Is it true that when you call "Close()" on the file, the subsequent "RecordAppend()" call always ends up in a new chunk? What's the case where the appended record goes to the end of the current chunk, but not to a new chunk?

File: wiki/QFS-Client-Reference.md [hunk 3]
```diff
@@ -283,3 +283,92 @@ _write-behind threshold_ by subsequent write calls or until user calls
...
+
+    Byte at offset 0 is 31
+    Byte at offset 1 is 32
+    Byte at offset 2 is 33
+    Read 3 bytes
+
+### Writes with O_APPEND
+If a file is opened with O_APPEND, then a Write behaves the same as a RecordAppend.
+
+To do a traditional append to end of file with a single writer, open the file for write then
```
Comment (mckurt) [inline wiki/QFS-Client-Reference.md]: This seems to be unrelated to "Writes with O_APPEND". Perhaps we need to put this under its own section. Something like "Seek and Write without O_APPEND" or "Traditional Append".

File: wiki/QFS-Client-Reference.md (hunk 2 above)
Comment (mikeov) [inline wiki/QFS-Client-Reference.md]: With append the answer is "no". Client close does not "close" the chunk, at most it invalidates / "closes" client's write ids.

Comment (mikeov) [inline wiki/QFS-Client-Reference.md]: Append supported *only* with replicated files. In other words, append is not supported with RS and object store files.

File: wiki/QFS-Client-Reference.md [hunk 4]
```diff
@@ -283,3 +283,92 @@ _write-behind threshold_ by subsequent write calls or until user calls
...
+    client->RecordAppend(fd, &data[1], 1);
+    client->RecordAppend(fd, &data[2], 1);
+    client->Close(fd);
+
+will result in a file with 2 chunks, each replicated 3 times, and with an size of
+134217728 bytes.
+
+An attempt to open this file for read will stall until all of the chunks are considered
+stable.
+
```
Comment (mikeov) [inline wiki/QFS-Client-Reference.md]: The file can be open for read anytime, however only "stable" chunks are available for read, as non stable chunks will either have write lease, or will be in transitioning to stable. One can think (and use) file opened for append as a message queue, and chunks as items in this queue. The once chunk transitions into stable state it can be read.
